### PR TITLE
KOGITO-8830: Add integration tests for expression autocompletion

### PR DESCRIPTION
**Pull request by tomasdavidorg**
https://issues.redhat.com/browse/KOGITO-8830

### CODE REVIEW

File: packages/serverless-logic-web-tools/it-tests/fixtures/expression/expression.sw.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        }
+      ],
+      "end": true
+    }
+  ],
+  "id": "Expression test",
+  "version": "0.1",
+  "specVersion": "0.8",
+  "start": "Expression state",
+  "dataInputSchema": "https://raw.githubusercontent.com/kiegroup/kogito-examples/stable/serverless-workflow-examples/serverless-workflow-expression-quarkus/src/main/resources/schema/expression.json",
```
Comment (paulovmr) [inline packages/serverless-logic-web-tools/it-tests/fixtures/expression/expression.sw.json]: @tomasdavidorg I think it is better to have a local copy of those linked files instead of a URL to avoid breaking the build for changes in the kogito-examples repository.

**Review by paulovmr — CHANGES_REQUESTED** [review]

File: packages/serverless-logic-web-tools/it-tests/e2e/Expression.cy.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        // create path to fromStateData value
+        var path = "";
+        for (var row = 0; row < 10; row++) {
+          path += "{downArrow}";
+        }
+        for (var column = 0; column < 30; column++) {
+          path += "{rightArrow}";
+        }
+
+        // move to value of actions -> actionDataFilter -> fromStateData and invoke content assist
```
Comment (barboras7) [inline packages/serverless-logic-web-tools/it-tests/e2e/Expression.cy.ts]: All code below this line till the end of the test method is the same as the code in the other test method. It could be reused instead of copy+pasting, wdyt?

File: packages/serverless-logic-web-tools/it-tests/fixtures/expression/expression.sw.json (hunk 1 above)
Comment (tomasdavidorg) [reply packages/serverless-logic-web-tools/it-tests/fixtures/expression/expression.sw.json]: @paulovmr The online editor does not support local files. I can do it by including the spec and schema files in the application itself and pointing to the `https://localhost:9020/it-tests/` url. It is not the best solution as the url is hard coded. Maybe I can add the url in the test by the `SERVERLESS_LOGIC_WEB_TOOLS__cypressUrl` property. 
WDYT about the packing by the webpack? Is it ok?

File: packages/serverless-logic-web-tools/it-tests/e2e/Expression.cy.ts (hunk 2 above)
Comment (tomasdavidorg) [reply packages/serverless-logic-web-tools/it-tests/e2e/Expression.cy.ts]: Cypress best practices are against using the separated functions. I would have to create custom command but this is designed for shorted peace of code which controls only some specific components and not replacing the testing code. 

I used parameterized tests instead. I think it looks better now. WDYT?

**Comment by tomasdavidorg** [comment]
Still WIP. I will try to improve the part where I use the hard-coded `localhost:9020` url.

File: packages/serverless-logic-web-tools/it-tests/fixtures/expression/expression.sw.json (hunk 1 above)
Comment (paulovmr) [reply packages/serverless-logic-web-tools/it-tests/fixtures/expression/expression.sw.json]: @tomasdavidorg In this case, I prefer to keep the URL to kogito-examples, but point to the latest tag instead of a branch, wdyt?

**Review by barboras7 — APPROVED** [review]
Looks good, thanks!

File: packages/serverless-logic-web-tools/it-tests/fixtures/expression/expression.sw.json (hunk 1 above)
Comment (tomasdavidorg) [reply packages/serverless-logic-web-tools/it-tests/fixtures/expression/expression.sw.json]: @paulovmr Make sense. Updated.

**Comment by tomasdavidorg** [comment]
@barboras7 @paulovmr Thanks for the reviews. There are two new changes:

1. `dataInputSchema` and `operation` urls are changed back to use files from kogito-examples repository but this time they use the latest tag: `1.36.0.Final`.
2.  For the navigation in textarea, I added new custom command `moveToPosition`.

Please take a look again. Thanks 🙂.

**Comment by barboras7** [comment]
The changes look good to me, thanks for the update @tomasdavidorg! :tada:

File: packages/vscode-extension-serverless-workflow-editor/package.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
     "pack:prod": "vsce package --githubBranch main --no-dependencies -o ./dist/vscode_extension_serverless_workflow_editor_$npm_package_version.vsix",
     "run:webmode": "pnpm vscode-test-web --browserType=chromium --extensionDevelopmentPath=. --version=stable",
     "test:it": "run-script-if --ignore-errors \"$(build-env integrationTests.ignoreFailures)\" --bool \"$(build-env integrationTests.run)\" --then \"pnpm test:it:clean\" \"cpr it-tests/resources it-tests-tmp/resources\" \"tsc --project tsconfig.it-tests.json --skipLibCheck --sourceMap true\" \"extest setup-and-run --yarn -c max -u -e ./test-resources -o ./it-tests/settings.json out/*.test.js\"",
-    "test:it:clean": "rimraf ./dist-it-tests && rimraf ./test-resources && rimraf ./out && rimraf ./it-tests-tmp && rimraf test-resou && rimraf *.vsix",
+    "test:it:clean": "rimraf ./dist-it-tests && rimraf ./test-resources && rimraf ./out && rimraf ./it-tests-tmp && rimraf test-resource && rimraf *.vsix",
```
Comment (caponetto) [inline packages/vscode-extension-serverless-workflow-editor/package.json]: Please see this https://github.com/kiegroup/kie-tools/pull/1547.
Can you please revert this change @tomasdavidorg?
This typo will be fixed in the other PR I've mentioned.

Comment (tomasdavidorg) [reply packages/vscode-extension-serverless-workflow-editor/package.json]: Reverted

**Review by caponetto — APPROVED** [review]

**Comment by caponetto** [comment]
Hi @barboras7.
If this PR is ok can you please give the final approval?

**Review by barboras7 — APPROVED** [review]

**Comment by caponetto** [comment]
Since @paulovmr set his review as "requested changes", I cannot merge until he approves. 
@paulovmr please take a look at this PR when possible :)

**Review by paulovmr — APPROVED** [review]
